src/lib/parse/contracts.js

```diff
@@ -0,0 +1,261 @@
+    if (!example) {
+        logger.error(`No request/response pairs found for: ${httpVerb} "${url}"`);
+        return;
+    }
+
+    const request = example.requests[0] && example.requests[0];
+    const response = example.responses[0] && example.responses[0];
+
+    return {
+        request: getSchema(request),
+        response: getSchema(response)
+    };
+};
+
+const getSchema = (body: Body): ?JsonSchema => {
+    if (!body) return;
+
+    const validatedBody: Body = specSchema.validateAndParseSchema(body);
+    return validatedBody.schema;
+}
+
+const removeInvalidActions = (resource: BlueprintResource, contractActions: ?Actions): BlueprintResource => {
+    const validActions: Array<BlueprintAction> = []
+
+    resource.actions.forEach((action) => {
+        const contractAction = contractActions && contractActions[action.method];
+        if (!contractAction) {
+            logger.error(`${action.method} ${resource.uriTemplate} is not in the contract`);
+            return;
+        }
+        if (!action.examples) return;
+        const validExamples: Array<Example> = [];
+
+        action.examples.forEach((example, exampleIndex) => {
+            const validRequests: Array<Body> = [];
+            const validResponses: Array<Body> = [];
+            if (action.method === 'POST' || action.method === 'PUT') {
+                if (example.requests.length !== example.responses.length) {
+                    throw new Error('Number of requests and responses are not equal');
+                }
+
+                for (let requestIndex = 0;requestIndex < example.requests.length; requestIndex++) {
+                    const request = example.requests[requestIndex];
+                    const reqBody = JSON.parse(request.body || "");
+                    const reqResult: SchemaValidationResult = specSchema.matchWithSchema(reqBody, contractAction.request);
+                    if (!reqResult.valid) {
+                        logger.error(`${action.method} ${resource.uriTemplate} example[${exampleIndex}] request[${requestIndex}] failed validation: \n\t${reqResult.niceErrors.join('\n\t')}`);
+                    }
+
+                    const response = example.responses[requestIndex];
+                    const respBody = JSON.parse(response.body || "");
+                    const respResult: SchemaValidationResult = specSchema.matchWithSchema(respBody, contractAction.response);
+                    if (!respResult.valid) {
+                        logger.error(`${action.method} ${resource.uriTemplate} example[${exampleIndex}] response[${requestIndex}] failed validation: \n\t${respResult.niceErrors.join('\n\t')}`);
+                    }
+
+                    if (reqResult.valid && respResult.valid) {
+                        validRequests.push(request);
+                        validResponses.push(response);
+                    }
+                }
+            } else {
+                example.responses.forEach((response, responseIndex) => {
```

Comment:
add comment to call out very visibly that this is where we validate GET/DELETE test fixtures.